### ISSUE TITLE
shouldn't there be a calcValidEmissions()?

### ISSUE THREAD

**Issue by 0UmfHxcvx5J7JoaOhFSs5mncnisTJJ6q** [pik-piam/moinput#19]
7bfeef5
```
  LU_EDGAR_LU    <- calcOutput(type="ValidEmissions", datasource="EDGAR_LU", aggregate=FALSE, try=TRUE)
  LU_CEDS        <- calcOutput(type="ValidEmissions", datasource="CEDS", aggregate=FALSE, try=TRUE)
  LU_FAO_EmisLUC <- calcOutput(type="ValidEmissions", datasource="FAO_EmisLUC", aggregate=FALSE, try=TRUE)
  LU_FAO_EmisAg  <- calcOutput(type="ValidEmissions", datasource="FAO_EmisAg", aggregate=FALSE, try=TRUE)
  LU_PRIMAPhist  <- calcOutput(type="ValidEmissions", datasource="PRIMAPhist", aggregate=FALSE, try=TRUE)
  LU_IPCC        <- calcOutput(type="ValidEmissions", datasource="IPCC", aggregate=FALSE, try=TRUE)
  LU_Nsurplus2   <- calcOutput(type="ValidEmissions", datasource="Nsurplus2", aggregate=FALSE, try=TRUE)
```

`/p/tmp/pehl/preprocessing-remind/log-14375708.out`
```
> Run calcOutput(type = "ValidEmissions", datasource = "EDGAR_LU",     aggregate = FALSE, try = TRUE)
Error: Type "ValidEmissions" is not a valid output type. Check getCalculations("calc") for available types!
```

**Comment by abhimishr** [comment]
calcValidEmissions exists. It is is a part of mrvalidation package.

**Comment by dklein-pik** [comment]
> calcValidEmissions exists. It is is a part of mrvalidation package.

Thanks, Abhi. I know. But I was not able to use it inside moinput without calling "library(mrvalidation)" explicitly, which I want to avoid.

**Comment by abhimishr** [comment]
Hi David, would it be possible to add a mrvalidation dependency in moinput?

Loading mrvalidation calls moinput but not the other way round for some reason.

**Comment by tscheypidi** [comment]
Two-way dependencies between packages are not possible for good reasons. That means you cannot use calcValidEmissions in moinput. However, moinput also contains some emission calculations (which are used in validemissions, for instance calcEmissionInventory. Does that help perhaps (I am not sure from the initial issue description what the actual purpose of this exercise is)?

**Comment by dklein-pik** [comment]
The purpose of the exercise: REMIND would like to include MAgPIE's historical emissions in our historical.mif to show them in validation plots. The historical.mif is calculated by calcHistorical (from moinput) and uses calcValidemissions (from mrvalidation).

**Comment by tscheypidi** [comment]
ok, in this case you need to move calcHistorical either to mrvalidation or to another package which is then based on mrvalidation and moinput

**Comment by 0UmfHxcvx5J7JoaOhFSs5mncnisTJJ6q** [comment]
Or suggest `mrvalidation` in `moinput`, and make calls to `calcValidEmissions()` and the like dependent on the presence of it, like we did in `remind::compareScenarios` [(click)](https://github.com/pik-piam/remind/blob/master/R/compareScenarios.R#L400), pursuant to [Writing R Extensions 1.1.3.1](https://cran.r-project.org/doc/manuals/R-exts.html#Suggested-packages).

**Comment by tscheypidi** [comment]
You still would get two-way dependencies. Hence, this is not a proper solution to the problem.

**Comment by Loisel** [comment]
+1

**Comment by tscheypidi** [comment]
fixed in c3ff4dbac7aed1694f41e76f15ccbf0193479833
